core/src/test/java/io/kestra/core/runners/pebble/filters/DistinctFilterTest.java

```diff
@@ -0,0 +1,97 @@
+package io.kestra.core.runners.pebble.filters;
+
+import com.google.common.collect.ImmutableMap;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.runners.VariableRenderer;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@KestraTest
+class DistinctFilterTest {
+
+    @Inject
+    VariableRenderer variableRenderer;
+
+    @Test
+    void toDistinctFilter() throws IllegalVariableEvaluationException {
+        ZonedDateTime date = ZonedDateTime.parse("2013-09-08T16:19:00+02").withZoneSameLocal(ZoneId.systemDefault());
+
+        ImmutableMap<String, Object> vars = ImmutableMap.of(
+            "vars", ImmutableMap.of("second", Map.of(
+                "string", "string",
+                "int", 1,
+                "float", 1.123F,
+                "list", Arrays.asList(
+                    "one", "two", "one", "three", "four", "five", "three",
+                    1, 2, 3, 1, 2, 2,
+                    1.123F, 2.123F, 1.123F, 10.000F, 10.000F
+                ),
+                "bool", true,
+                "date", date,
+                "map", Map.of(
+                    "string", "string",
+                    "int", 1,
+                    "float", 1.123F
+                )
+            ))
+        );
+
+        //Test rendering the list without the distinct filter
+        String render = variableRenderer.render("{{ vars.second.list }}", vars);
+        
+        //Verify that the list contains duplicates
+        assertThat(render, containsString("one"));
+        assertThat(render, containsString("two"));
+        assertThat(render, containsString("three"));
+        assertThat(render, containsString("1"));
+        assertThat(render, containsString("1.123"));
+
+        //Apply the distinct filter
+        String distinctRender = variableRenderer.render("{{ vars.second.list | distinct }}", vars);
+		
+        //Verify that duplicates are removed from the list
+        assertThat(distinctRender, is("[\"one\",\"two\",\"three\",\"four\",\"five\",1,2,3,1.123,2.123,10.0]"));
+        assertThat(distinctRender, not(containsString("one,one"))); //Ensure duplicates are removed
+        assertThat(distinctRender, startsWith("["));
+        assertThat(distinctRender, endsWith("]"));
+
+        //Edge case: an empty list
+        render = variableRenderer.render("{{ [] | distinct }}", Map.of());
+        assertThat(render, is("[]"));
+    }
+
+    @Test
+    void distinctFilterWithNullInput() {
```

Comment:
I think this test will now pass.

Reply:
I was not clear, this test would not pass as null is allowed.
Can you update the test so we can merge?